plasmapy/physics/transport.py

```diff
@@ -0,0 +1,172 @@
+
+    for particle, i in zip(particles, range(2)):
+
+        try:
+            masses[i] = ion_mass(particles[i])
+        except Exception:
+            raise ValueError("Unable to find mass of particle: " +
+                             str(particles[i]) + " in Coulomb_logarithm.")
+
+        try:
+            charges[i] = np.abs(e*charge_state(particles[i]))
+            if charges[i] is None:
+                raise
+        except Exception:
+            raise ValueError("Unable to find charge of particle: " +
+                             str(particles[i]) + " in Coulomb_logarithm.")
+
+    reduced_mass = masses[0] * masses[1] / (masses[0] + masses[1])
+
+    # The outer impact parameter is the Debye length.  At distances
+    # greater than the Debye length, the electrostatic potential of a
+    # single particle is screened out by the electrostatic potentials
+    # of other particles.  Past this distance, the electric fields of
+    # individual particles do not affect each other much.  This
+    # expression neglects screening by heavier ions.
+
+    b_max = Debye_length(T, n_e)
+
+    # The choice of inner impact parameter is more controversial.
+    # There are two broad possibilities, and the expressions in the
+    # literature often differ by factors of order unity or by
+    # interchanging the reduced mass with the test particle mass.
+
+    # The relative velocity is a source of uncertainty.  It is
+    # reasonable to make an assumption relating the thermal energy to
+    # the kinetic energy: reduced_mass*velocity**2 is approximately
+    # equal to 3*k_B*T.
+
+    # If no relative velocity is inputted, then we make an assumption
+    # that relates the thermal energy to the kinetic energy:
+    # reduced_mass*velocity**2 is approximately equal to 3*k_B*T.
+
+    if V is None:
+        V = np.sqrt(3 * k_B * T / reduced_mass)
+
+    # The first possibility is that the inner impact parameter
+    # corresponds to a deflection of 90 degrees, which is valid when
+    # classical effects dominate.
+
+    b_perp = charges[0] * charges[1] / (4 * pi * eps0 * reduced_mass * V**2)
+
+    # The second possibility is that the inner impact parameter is a
+    # de Broglie wavelength.  There remains some ambiguity as to which
+    # mass to choose to go into the de Broglie wavelength calculation.
+    # Here we use the reduced mass, which will be of the same order as
+    # mass of the smaller particle and thus the longer de Broglie
+    # wavelength.
+
+    b_deBroglie = hbar / (2 * reduced_mass * V)
+
+    # Coulomb-style collisions will not happen for impact parameters
+    # shorter than either of these two impact parameters, so we choose
+    # the larger of these two possibilities.
+
+    if b_perp > b_deBroglie:
+        b_min = b_perp
+    else:
+        b_min = b_deBroglie
+
+    # Now that we know how many approximations have to go into plasma
+    # transport theory, we shall celebrate by returning the Coulomb
+    # logarithm.
```

Comment:
And this is where @namurphy joins the many unsung heroes of transport theory, sitting at the right hand of Braginskii. Kudos!